internal/runtimehandlerhooks/high_performance_hooks_linux.go

```diff
@@ -724,6 +725,17 @@ func (h *HighPerformanceHooks) setIRQLoadBalancing(ctx context.Context, c *oci.C
 	h.updateIRQSMPAffinityLock.Lock()
 	defer h.updateIRQSMPAffinityLock.Unlock()
 
+	if !enable {
+		// If we're disabling IRQ SMP Affinity, save the container ID
+		h.irqSMPAffinityDisabledSet[c.ID()] = struct{}{}
+	} else if _, ok := h.irqSMPAffinityDisabledSet[c.ID()]; ok {
+		// If we're reenabling, and we haven't already reenabled, delete from the set
+		delete(h.irqSMPAffinityDisabledSet, c.ID())
+	} else {
+		// Otherwise, skip the operation to not redo the cleanup
+		return nil
+	}
+
```

Comment:
nit(non blocking): we should make the whole hooks idempotent, not only this function.

Reply:
most of them don't access shared state like this, so narrowing the scope will be more efficient. That's a good callout if we need more state coordination though